From a035729db6d0ded4a12d7c34b98773ff075d6293 Mon Sep 17 00:00:00 2001
From: Jackson West <jgw@amazon.com>
Date: Tue, 3 Jan 2023 20:04:02 -0800
Subject: [PATCH] Bumps go version in go.mod

When deps are downloaded with `go mod vendor` which our
build system does for all builds, go build will get pretty
picky about the version specified in the go.mod. The version
is meant to be the minimum version needed to build the package.
This project is using 1.17+ golang features now.

Reference: https://stackoverflow.com/questions/58204618/what-are-the-implications-of-using-the-go-version-directive-within-a-go-module
Signed-off-by: Jackson West <jgw@amazon.com>
---
 go.mod | 188 ++++++++++++++++++++++++++++++++++++++++++---------------
 1 file changed, 139 insertions(+), 49 deletions(-)

diff --git a/go.mod b/go.mod
index 2e959d6e..64b7323d 100644
--- a/go.mod
+++ b/go.mod
@@ -1,57 +1,147 @@
 module sigs.k8s.io/aws-iam-authenticator

-go 1.16
+go 1.17

 require (
-    github.com/aws/aws-sdk-go v1.44.107
-    github.com/fsnotify/fsnotify v1.4.9
-    github.com/gofrs/flock v0.7.0
-    github.com/manifoldco/promptui v0.9.0
-    github.com/onsi/ginkgo v1.14.0
-    github.com/onsi/gomega v1.10.1
-    github.com/prometheus/client_golang v1.11.0
-    github.com/sirupsen/logrus v1.8.1
-    github.com/spf13/cobra v1.1.3
-    github.com/spf13/viper v1.7.0
-    golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
-    gopkg.in/yaml.v2 v2.4.0
-    k8s.io/api v0.22.1
-    k8s.io/apimachinery v0.22.1
-    k8s.io/client-go v0.22.1
-    k8s.io/code-generator v0.22.1
-    k8s.io/component-base v0.22.1
-    k8s.io/kubernetes v1.22.0
-    k8s.io/sample-controller v0.22.1
-    sigs.k8s.io/yaml v1.2.0
+	github.com/aws/aws-sdk-go v1.44.107
+	github.com/fsnotify/fsnotify v1.4.9
+	github.com/gofrs/flock v0.7.0
+	github.com/manifoldco/promptui v0.9.0
+	github.com/onsi/ginkgo v1.14.0
+	github.com/onsi/gomega v1.10.1
+	github.com/prometheus/client_golang v1.11.0
+	github.com/sirupsen/logrus v1.8.1
+	github.com/spf13/cobra v1.1.3
+	github.com/spf13/viper v1.7.0
+	golang.org/x/time v0.0.0-20210723032227-1f47c861a9ac
+	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/api v0.22.1
+	k8s.io/apimachinery v0.22.1
+	k8s.io/client-go v0.22.1
+	k8s.io/code-generator v0.22.1
+	k8s.io/component-base v0.22.1
+	k8s.io/kubernetes v1.22.0
+	k8s.io/sample-controller v0.22.1
+	sigs.k8s.io/yaml v1.2.0
+)
+
+require (
+	github.com/PuerkitoBio/purell v1.1.1 // indirect
+	github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578 // indirect
+	github.com/beorn7/perks v1.0.1 // indirect
+	github.com/blang/semver v3.5.1+incompatible // indirect
+	github.com/cespare/xxhash/v2 v2.1.1 // indirect
+	github.com/chzyer/readline v0.0.0-20180603132655-2972be24d48e // indirect
+	github.com/cyphar/filepath-securejoin v0.2.2 // indirect
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/docker/distribution v2.7.1+incompatible // indirect
+	github.com/emicklei/go-restful v2.9.5+incompatible // indirect
+	github.com/evanphx/json-patch v4.11.0+incompatible // indirect
+	github.com/felixge/httpsnoop v1.0.1 // indirect
+	github.com/go-logr/logr v0.4.0 // indirect
+	github.com/go-openapi/jsonpointer v0.19.5 // indirect
+	github.com/go-openapi/jsonreference v0.19.5 // indirect
+	github.com/go-openapi/swag v0.19.14 // indirect
+	github.com/gogo/protobuf v1.3.2 // indirect
+	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
+	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/google/go-cmp v0.5.5 // indirect
+	github.com/google/gofuzz v1.1.0 // indirect
+	github.com/google/uuid v1.1.2 // indirect
+	github.com/googleapis/gnostic v0.5.5 // indirect
+	github.com/grpc-ecosystem/grpc-gateway v1.16.0 // indirect
+	github.com/hashicorp/hcl v1.0.0 // indirect
+	github.com/imdario/mergo v0.3.5 // indirect
+	github.com/inconshreveable/mousetrap v1.0.0 // indirect
+	github.com/jmespath/go-jmespath v0.4.0 // indirect
+	github.com/josharian/intern v1.0.0 // indirect
+	github.com/json-iterator/go v1.1.11 // indirect
+	github.com/magiconair/properties v1.8.1 // indirect
+	github.com/mailru/easyjson v0.7.6 // indirect
+	github.com/matttproud/golang_protobuf_extensions v1.0.2-0.20181231171920-c182affec369 // indirect
+	github.com/mitchellh/mapstructure v1.1.2 // indirect
+	github.com/moby/spdystream v0.2.0 // indirect
+	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
+	github.com/modern-go/reflect2 v1.0.1 // indirect
+	github.com/nxadm/tail v1.4.4 // indirect
+	github.com/opencontainers/go-digest v1.0.0 // indirect
+	github.com/opencontainers/runc v1.0.1 // indirect
+	github.com/pelletier/go-toml v1.2.0 // indirect
+	github.com/pkg/errors v0.9.1 // indirect
+	github.com/prometheus/client_model v0.2.0 // indirect
+	github.com/prometheus/common v0.26.0 // indirect
+	github.com/prometheus/procfs v0.6.0 // indirect
+	github.com/spf13/afero v1.2.2 // indirect
+	github.com/spf13/cast v1.3.0 // indirect
+	github.com/spf13/jwalterweatherman v1.0.0 // indirect
+	github.com/spf13/pflag v1.0.5 // indirect
+	github.com/subosito/gotenv v1.2.0 // indirect
+	go.opentelemetry.io/contrib v0.20.0 // indirect
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.20.0 // indirect
+	go.opentelemetry.io/otel v0.20.0 // indirect
+	go.opentelemetry.io/otel/exporters/otlp v0.20.0 // indirect
+	go.opentelemetry.io/otel/metric v0.20.0 // indirect
+	go.opentelemetry.io/otel/sdk v0.20.0 // indirect
+	go.opentelemetry.io/otel/sdk/export/metric v0.20.0 // indirect
+	go.opentelemetry.io/otel/sdk/metric v0.20.0 // indirect
+	go.opentelemetry.io/otel/trace v0.20.0 // indirect
+	go.opentelemetry.io/proto/otlp v0.7.0 // indirect
+	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83 // indirect
+	golang.org/x/mod v0.4.2 // indirect
+	golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd // indirect
+	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d // indirect
+	golang.org/x/sys v0.0.0-20211216021012-1d35b9e2eb4e // indirect
+	golang.org/x/term v0.0.0-20210927222741-03fcf44c2211 // indirect
+	golang.org/x/text v0.3.7 // indirect
+	golang.org/x/tools v0.1.2 // indirect
+	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
+	google.golang.org/appengine v1.6.5 // indirect
+	google.golang.org/genproto v0.0.0-20210602131652-f16073e35f0c // indirect
+	google.golang.org/grpc v1.38.0 // indirect
+	google.golang.org/protobuf v1.26.0 // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/ini.v1 v1.51.0 // indirect
+	gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+	k8s.io/apiserver v0.22.0 // indirect
+	k8s.io/component-helpers v0.22.0 // indirect
+	k8s.io/gengo v0.0.0-20201214224949-b6c5ce23f027 // indirect
+	k8s.io/klog/v2 v2.9.0 // indirect
+	k8s.io/kube-openapi v0.0.0-20210421082810-95288971da7e // indirect
+	k8s.io/kubectl v0.0.0 // indirect
+	k8s.io/kubelet v0.0.0 // indirect
+	k8s.io/utils v0.0.0-20210707171843-4b05e18ac7d9 // indirect
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.22 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.1.2 // indirect
 )

 replace (
-    k8s.io.client-go => k8s.io/client-go v0.20.0
-    k8s.io/api => k8s.io/api v0.22.0
-    k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.22.0
-    k8s.io/apimachinery => k8s.io/apimachinery v0.23.0-alpha.0
-    k8s.io/apiserver => k8s.io/apiserver v0.22.0
-    k8s.io/cli-runtime => k8s.io/cli-runtime v0.22.0
-    k8s.io/client-go => k8s.io/client-go v0.22.0
-    k8s.io/cloud-provider => k8s.io/cloud-provider v0.22.0
-    k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.22.0
-    k8s.io/code-generator => k8s.io/code-generator v0.22.2-rc.0
-    k8s.io/component-base => k8s.io/component-base v0.22.0
-    k8s.io/component-helpers => k8s.io/component-helpers v0.22.0
-    k8s.io/controller-manager => k8s.io/controller-manager v0.22.0
-    k8s.io/cri-api => k8s.io/cri-api v0.23.0-alpha.0
-    k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.22.0
-    k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.22.0
-    k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.22.0
-    k8s.io/kube-proxy => k8s.io/kube-proxy v0.22.0
-    k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.22.0
-    k8s.io/kubectl => k8s.io/kubectl v0.22.0
-    k8s.io/kubelet => k8s.io/kubelet v0.22.0
-    k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.22.0
-    k8s.io/metrics => k8s.io/metrics v0.22.0
-    k8s.io/mount-utils => k8s.io/mount-utils v0.22.1-rc.0
-    k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.22.0
-    k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.22.0
-    k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.22.0
-    k8s.io/sample-controller => k8s.io/sample-controller v0.22.0
+	k8s.io.client-go => k8s.io/client-go v0.20.0
+	k8s.io/api => k8s.io/api v0.22.0
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.22.0
+	k8s.io/apimachinery => k8s.io/apimachinery v0.23.0-alpha.0
+	k8s.io/apiserver => k8s.io/apiserver v0.22.0
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.22.0
+	k8s.io/client-go => k8s.io/client-go v0.22.0
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.22.0
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.22.0
+	k8s.io/code-generator => k8s.io/code-generator v0.22.2-rc.0
+	k8s.io/component-base => k8s.io/component-base v0.22.0
+	k8s.io/component-helpers => k8s.io/component-helpers v0.22.0
+	k8s.io/controller-manager => k8s.io/controller-manager v0.22.0
+	k8s.io/cri-api => k8s.io/cri-api v0.23.0-alpha.0
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.22.0
+	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.22.0
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.22.0
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.22.0
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.22.0
+	k8s.io/kubectl => k8s.io/kubectl v0.22.0
+	k8s.io/kubelet => k8s.io/kubelet v0.22.0
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.22.0
+	k8s.io/metrics => k8s.io/metrics v0.22.0
+	k8s.io/mount-utils => k8s.io/mount-utils v0.22.1-rc.0
+	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.22.0
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.22.0
+	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.22.0
+	k8s.io/sample-controller => k8s.io/sample-controller v0.22.0
 )
--
2.39.0

